code/modules/mob/living/simple_animal/hostile/megafauna/demonic_frost_miner.dm

```diff
@@ -204,12 +204,30 @@ Difficulty: Extremely Hard
 	homing_turn_speed = 3
 	damage_type = BURN
 
-/obj/projectile/colossus/frost_orb/on_hit(atom/target, blocked = 0, pierce_hit)
+/obj/projectile/frost_miner/can_hit_target(atom/target, direct_target = FALSE, ignore_loc = FALSE, cross_failed = FALSE)
+	if(isliving(target))
+		direct_target = TRUE
+	return ..(target, direct_target, ignore_loc, cross_failed)
+
+/obj/projectile/frost_miner/on_hit(atom/target, blocked = 0, pierce_hit)
 	. = ..()
+	if(isliving(target))
+		var/mob/living/dead_mob = target
+		if(dead_mob.stat == DEAD)
+			if(dead_mob.has_status_effect(/datum/status_effect/gutted))
+				return BULLET_ACT_FORCE_PIERCE
+			else
+				if(ismegafauna(firer))
+					var/mob/living/simple_animal/hostile/megafauna/megafauna = firer
+					megafauna.devour(target)
+		return
```

Comment:
this is some insanely bad indentation. you can definitely add some early returns in here